src/main/java/org/terasology/gf/tree/lsystem/AdvancedLSystemTreeDefinition.java

```diff
@@ -342,9 +345,7 @@ private TreeStructure generateTreeFromAxion(Vector3i location, String currentAxi
 
         BranchLocation branchLocation = treeStructure.getRootBranch();
         Vector3f position = location.toVector3f();
-        Matrix4f rotation = new Matrix4f();
-        rotation.setIdentity();
-        rotation.rotY(treeRotation);
+        Matrix4f rotation = new Matrix4f(new Quat4f(new Vector3f(0, 1, 0), treeRotation), BaseVector3f.ZERO, 1.0f);
```

Comment:
Hehe, I see what you mean with more constructors possibly being handy. But if it works it works :D